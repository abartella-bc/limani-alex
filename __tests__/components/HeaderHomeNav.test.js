/*
Copyright 2023 BlueCat Networks Inc.

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

import { shallow } from 'enzyme';
import HeaderHomeNav from '../../src/header/HeaderHomeNav';
import usePlatformData from '../../src/hooks/usePlatformData';

jest.unmock('../../src/header/HeaderHomeNav');

const mockUsePlatformDataValue = {
    data: jest.fn(),
    isFetchingData: false,
    triggerFetchData: jest.fn(),
};

jest.mock('../../src/hooks/usePlatformData', () =>
    jest.fn(() => {
        return mockUsePlatformDataValue;
    }),
);

const customPagePlatformData = {
    user: {
        home_url: 'custom/navigation',
    },
};

describe('HeaderHomeNav', () => {
    describe('Rendering', () => {
        it('Render HeaderHomeNav component with default props', () => {
            window.location = new URL('https://gateway-test.com');
            const wrapper = shallow(<HeaderHomeNav />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderHomeNav component with props', () => {
            window.location = new URL('https://gateway-test.com');
            const wrapper = shallow(<HeaderHomeNav className='varClassName' />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderHomeNav component with inactive Home', () => {
            window.location = new URL('https://gateway-test.com/subpage');
            const wrapper = shallow(<HeaderHomeNav className='varClassName' />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderHomeNav component with active Home', () => {
            window.location = new URL(
                'https://gateway-test.com/custom/navigation',
            );
            usePlatformData.mockReturnValue({
                data: customPagePlatformData,
                isFetchingData: false,
                triggerFetchData: jest.fn(),
            });
            const wrapper = shallow(<HeaderHomeNav className='varClassName' />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });

        it('Render HeaderHomeNav component with active Workflows', () => {
            window.location = new URL('https://gateway-test.com/index');
            usePlatformData.mockReturnValue({
                data: customPagePlatformData,
                isFetchingData: false,
                triggerFetchData: jest.fn(),
            });
            const wrapper = shallow(<HeaderHomeNav className='varClassName' />);
            expect(wrapper.getElement()).toMatchSnapshot();
        });
    });
});
