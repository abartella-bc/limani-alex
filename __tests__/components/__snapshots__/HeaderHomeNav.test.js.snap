// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderHomeNav Rendering Render HeaderHomeNav component with active Home 1`] = `
<div
  className="HeaderHomeNav varClassName"
>
  <a
    aria-current="page"
    className="HeaderHomeNav__link"
    href="/home"
    id="headerHomeLink"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [
              63499,
              "home-lg",
            ],
            "e3af",
            "M543.8 287.6c17 0 32-14 32-32.1c1-9-3-17-11-24L512 185V64c0-17.7-14.3-32-32-32H448c-17.7 0-32 14.3-32 32v36.7L309.5 7c-6-5-14-7-21-7s-15 1-22 8L10 231.5c-7 7-10 15-10 24c0 18 14 32.1 32 32.1h32v69.7c-.1 .9-.1 1.8-.1 2.8V472c0 22.1 17.9 40 40 40h16c1.2 0 2.4-.1 3.6-.2c1.5 .1 3 .2 4.5 .2H160h24c22.1 0 40-17.9 40-40V448 384c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v64 24c0 22.1 17.9 40 40 40h24 32.5c1.4 0 2.8 0 4.2-.1c1.1 .1 2.2 .1 3.3 .1h16c22.1 0 40-17.9 40-40V455.8c.3-2.6 .5-5.3 .5-8.1l-.7-160.2h32z",
          ],
          "iconName": "house-chimney",
          "prefix": "fas",
        }
      }
      label="Home"
    />
  </a>
  <a
    aria-current={null}
    className="HeaderHomeNav__link"
    href="/index"
    id="headerWorkflowsLink"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [
              128470,
            ],
            "f2d0",
            "M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zM96 96H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H96c-17.7 0-32-14.3-32-32s14.3-32 32-32z",
          ],
          "iconName": "window-maximize",
          "prefix": "fas",
        }
      }
      label="Workflows"
    />
  </a>
</div>
`;

exports[`HeaderHomeNav Rendering Render HeaderHomeNav component with active Workflows 1`] = `
<div
  className="HeaderHomeNav varClassName"
>
  <a
    aria-current={null}
    className="HeaderHomeNav__link"
    href="/home"
    id="headerHomeLink"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [
              63499,
              "home-lg",
            ],
            "e3af",
            "M543.8 287.6c17 0 32-14 32-32.1c1-9-3-17-11-24L512 185V64c0-17.7-14.3-32-32-32H448c-17.7 0-32 14.3-32 32v36.7L309.5 7c-6-5-14-7-21-7s-15 1-22 8L10 231.5c-7 7-10 15-10 24c0 18 14 32.1 32 32.1h32v69.7c-.1 .9-.1 1.8-.1 2.8V472c0 22.1 17.9 40 40 40h16c1.2 0 2.4-.1 3.6-.2c1.5 .1 3 .2 4.5 .2H160h24c22.1 0 40-17.9 40-40V448 384c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v64 24c0 22.1 17.9 40 40 40h24 32.5c1.4 0 2.8 0 4.2-.1c1.1 .1 2.2 .1 3.3 .1h16c22.1 0 40-17.9 40-40V455.8c.3-2.6 .5-5.3 .5-8.1l-.7-160.2h32z",
          ],
          "iconName": "house-chimney",
          "prefix": "fas",
        }
      }
      label="Home"
    />
  </a>
  <a
    aria-current="page"
    className="HeaderHomeNav__link"
    href="/index"
    id="headerWorkflowsLink"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [
              128470,
            ],
            "f2d0",
            "M64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64zM96 96H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H96c-17.7 0-32-14.3-32-32s14.3-32 32-32z",
          ],
          "iconName": "window-maximize",
          "prefix": "fas",
        }
      }
      label="Workflows"
    />
  </a>
</div>
`;

exports[`HeaderHomeNav Rendering Render HeaderHomeNav component with default props 1`] = `
<div
  className="HeaderHomeNav"
>
  <a
    aria-current="page"
    className="HeaderHomeNav__link"
    href="/home"
    id="headerHomeLink"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [
              63499,
              "home-lg",
            ],
            "e3af",
            "M543.8 287.6c17 0 32-14 32-32.1c1-9-3-17-11-24L512 185V64c0-17.7-14.3-32-32-32H448c-17.7 0-32 14.3-32 32v36.7L309.5 7c-6-5-14-7-21-7s-15 1-22 8L10 231.5c-7 7-10 15-10 24c0 18 14 32.1 32 32.1h32v69.7c-.1 .9-.1 1.8-.1 2.8V472c0 22.1 17.9 40 40 40h16c1.2 0 2.4-.1 3.6-.2c1.5 .1 3 .2 4.5 .2H160h24c22.1 0 40-17.9 40-40V448 384c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v64 24c0 22.1 17.9 40 40 40h24 32.5c1.4 0 2.8 0 4.2-.1c1.1 .1 2.2 .1 3.3 .1h16c22.1 0 40-17.9 40-40V455.8c.3-2.6 .5-5.3 .5-8.1l-.7-160.2h32z",
          ],
          "iconName": "house-chimney",
          "prefix": "fas",
        }
      }
      label="Home"
    />
  </a>
</div>
`;

exports[`HeaderHomeNav Rendering Render HeaderHomeNav component with inactive Home 1`] = `
<div
  className="HeaderHomeNav varClassName"
>
  <a
    aria-current={null}
    className="HeaderHomeNav__link"
    href="/home"
    id="headerHomeLink"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [
              63499,
              "home-lg",
            ],
            "e3af",
            "M543.8 287.6c17 0 32-14 32-32.1c1-9-3-17-11-24L512 185V64c0-17.7-14.3-32-32-32H448c-17.7 0-32 14.3-32 32v36.7L309.5 7c-6-5-14-7-21-7s-15 1-22 8L10 231.5c-7 7-10 15-10 24c0 18 14 32.1 32 32.1h32v69.7c-.1 .9-.1 1.8-.1 2.8V472c0 22.1 17.9 40 40 40h16c1.2 0 2.4-.1 3.6-.2c1.5 .1 3 .2 4.5 .2H160h24c22.1 0 40-17.9 40-40V448 384c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v64 24c0 22.1 17.9 40 40 40h24 32.5c1.4 0 2.8 0 4.2-.1c1.1 .1 2.2 .1 3.3 .1h16c22.1 0 40-17.9 40-40V455.8c.3-2.6 .5-5.3 .5-8.1l-.7-160.2h32z",
          ],
          "iconName": "house-chimney",
          "prefix": "fas",
        }
      }
      label="Home"
    />
  </a>
</div>
`;

exports[`HeaderHomeNav Rendering Render HeaderHomeNav component with props 1`] = `
<div
  className="HeaderHomeNav varClassName"
>
  <a
    aria-current="page"
    className="HeaderHomeNav__link"
    href="/home"
    id="headerHomeLink"
  >
    <HeaderIcon
      icon={
        Object {
          "icon": Array [
            576,
            512,
            Array [
              63499,
              "home-lg",
            ],
            "e3af",
            "M543.8 287.6c17 0 32-14 32-32.1c1-9-3-17-11-24L512 185V64c0-17.7-14.3-32-32-32H448c-17.7 0-32 14.3-32 32v36.7L309.5 7c-6-5-14-7-21-7s-15 1-22 8L10 231.5c-7 7-10 15-10 24c0 18 14 32.1 32 32.1h32v69.7c-.1 .9-.1 1.8-.1 2.8V472c0 22.1 17.9 40 40 40h16c1.2 0 2.4-.1 3.6-.2c1.5 .1 3 .2 4.5 .2H160h24c22.1 0 40-17.9 40-40V448 384c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v64 24c0 22.1 17.9 40 40 40h24 32.5c1.4 0 2.8 0 4.2-.1c1.1 .1 2.2 .1 3.3 .1h16c22.1 0 40-17.9 40-40V455.8c.3-2.6 .5-5.3 .5-8.1l-.7-160.2h32z",
          ],
          "iconName": "house-chimney",
          "prefix": "fas",
        }
      }
      label="Home"
    />
  </a>
</div>
`;
